Std/Data/HashMap/Basic.lean

```diff
@@ -88,26 +88,26 @@ def empty (capacity := 0) : Imp α β :=
     else ⟨nbuckets, Nat.zero_lt_of_ne_zero h⟩
   empty' n n.2
 
-/-- Calculates the bucket index from a hash value `u`. -/
-def mkIdx {n : Nat} (h : 0 < n) (u : USize) : {u : USize // u.toNat < n} :=
-  ⟨u % n, USize.modn_lt _ h⟩
+/-- Calculates the bucket index for the key `a`. -/
+def mkIdx [Hashable α] {n : Nat} (h : 0 < n) (a : α) : {u : USize // u.toNat < n} :=
+  ⟨(hash a |>.toUSize) % n, USize.modn_lt _ h⟩
```

Comment:
This change is performance sensitive, I would prefer not to change it without benchmarking first (and we don't have the setup to do that right now, but at least some evidence that the generated IR is not adversely affected would be good). In particular, the previous function was not monomorphized but the new one is.

Also, I believe that lean4 has since changed this part of the `Lean.HashMap` implementation a bit, and we might want to consider porting that change to std as well. (I think it doesn't change the defeqs so hopefully the proofs are unaffected...)

Reply:
Indeed it was, but it [was found](https://github.com/leanprover/lean4/commit/25beba66242118eed7857cf3666c43f74ba7d7cb) to "make no difference in practice". Do you think it's still worth porting then?